tests/functional/modules/test_zos_job_submit_func.py

```diff
@@ -706,3 +752,40 @@ def test_negative_job_submit_local_jcl_typrun_scan(ansible_zos_module):
         assert re.search(r'error ? ?', repr(result.get("msg")))
         assert result.get("jobs")[0].get("job_id") is not None
         assert result.get("jobs")[0].get("ret_code").get("msg_text") == "?"
+
+
+# This test case is related to the following GitHub issues:
+# - https://github.com/ansible-collections/ibm_zos_core/issues/677
+# - https://github.com/ansible-collections/ibm_zos_core/issues/972
+# - https://github.com/ansible-collections/ibm_zos_core/issues/1160
+# - https://github.com/ansible-collections/ibm_zos_core/issues/1255
+def test_zoau_bugfix_invalid_utf8_chars(ansible_zos_module):
+    try:
+        hosts = ansible_zos_module
+
+        # Copy C source and compile it.
+        hosts.all.file(path=TEMP_PATH, state="directory")
+        hosts.all.shell(
+            cmd="echo {0} > {1}/noprint.c".format(quote(C_SRC_INVALID_UTF8), TEMP_PATH)
+        )
+        hosts.all.shell(cmd="xlc -o {0}/noprint {0}/noprint.c")
+
+        # Create local JCL and submit it.
+        tmp_file = tempfile.NamedTemporaryFile(delete=True)
+        with open(tmp_file.name, "w") as f:
```

Comment:
Not as a change just as a question witch form is better doing a `open as f` or a `echo`

Reply:
For local files, `open as f`. For remote files, `echo` so we can avoid a call to our own collection